examples/openedx_courses/openedx_courses.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/asciimoo/colly"
+)
+
+type Course struct {
+	CourseID  string
+	Run       string
+	Name      string
+	Number    string
+	StartDate *time.Time
+	EndDate   *time.Time
+	URL       string
+}
+
+func main() {
+	// Instantiate default collector
+	c := colly.NewCollector()
+	// Using IndonesiaX as sample
+	c.AllowedDomains = []string{"indonesiax.co.id", "www.indonesiax.co.id"}
+
+	// Cache responses to prevent multiple download of pages
+	// even if the collector is restarted
+	c.CacheDir = "./cache"
+
+	courses := make([]Course, 0, 200)
+
+	// On every a element which has href attribute call callback
+	c.OnHTML("a[href]", func(e *colly.HTMLElement) {
+		link := e.Attr("href")
+		if !strings.HasPrefix(link, "/courses/") {
+			return
+		}
+		// start scaping the page under the link found
+		e.Request.Visit(link)
+	})
+
+	c.OnHTML("div[class=content-wrapper]", func(e *colly.HTMLElement) {
+		if e.DOM.Find("section.course-info").Length() == 0 {
+			return
+		}
+		title := strings.Split(e.ChildText(".course-title"), "\n")[0]
+		//provider := e.DOM.Find(".course-title > a").First()
+		course_id := e.DOM.Find("input[name=course_id]").First().AttrOr("value", "")
```

Comment:
This can be simplified to `course_id := e.ChildValue("input[name=course_id]", "value")`